pkg/controller/egress/controller.go

```diff
@@ -241,6 +243,20 @@ func (c *EgressController) setIPAllocation(egressName string, ip net.IP, poolNam
 	}
 }
 
+func pingEgressIP(egressIP string) (bool, error) {
+	cmd := exec.Command("ping", "-c", "3", "-w", "5", egressIP)
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return false, fmt.Errorf("ping command failed: %v. output: %s", err, string(output))
+	}
+
+	if strings.Contains(string(output), "3 packets transmitted, 3 received") {
+		return true, nil
+	}
+	return false, nil
+}
+
```

Comment:
ping is not available by default in image, calling commands like ping could potentially introduce performance issues or block your program if many Egress IPs are being checked !

Also the ping command and its options can differ across operating systems. For example, some systems use -t instead of -w for specifying timeout, and packet counts might behave differently. If you're running this across different environments, consider platform-specific behavior.